x/capability/keeper/keeper.go

```diff
@@ -235,14 +268,38 @@ func (sk ScopedKeeper) ReleaseCapability(ctx sdk.Context, cap *types.Capability)
 // by name. The module is not allowed to retrieve capabilities which it does not
 // own.
 func (sk ScopedKeeper) GetCapability(ctx sdk.Context, name string) (*types.Capability, bool) {
-	cap := sk.capStore.GetCapability(sk.module, name)
+	memStore := ctx.KVStore(sk.memKey)
+
+	key := types.RevCapabilityKey(sk.module, name)
+	indexBytes := memStore.Get(key)
+	index := sdk.BigEndianToUint64(indexBytes)
+	if len(indexBytes) == 0 {
+		// If a tx failed and NewCapability got reverted, it is possible
```

Comment:
Hmm. This is alright, although it would be a bit nicer to be able to update the map on tx reversion (we should know the pending changes by inspecting the changes to the mem store), is that too tricky? I think this is semantically OK.

Reply:
I don't know how we would do that, since only baseapp knows a tx got reverted and handles it. Don't think baseapp should be responsible for deleting entries from a go-map in capability. It wouldn't know how to do it.

Think that's too tricky, so have to deal with this indirection